week3/week3_hw/Program.cs

```diff
@@ -0,0 +1,188 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "week-3 homework!");
+
+app.MapGet("/account", () =>
+{
+  var account = new Account(100);
+  account.Deposit(100);
+  Console.WriteLine(System.Text.Json.JsonSerializer.Serialize(account));
+  Console.WriteLine($"Account balance is {account.Balance}");
+  account.Withdraw(10);
+  Console.WriteLine(System.Text.Json.JsonSerializer.Serialize(account));
+  Console.WriteLine($"Account new balance is {account.Balance}");
+  Console.WriteLine(System.Text.Json.JsonSerializer.Serialize(account));
+  // account.Withdraw(200); // ❌ we should not be able to withdraw more than we have in the balance
+});
+
+app.MapGet("/interface", () =>
+{
+  IAnimal dog = new Dog();
+  dog.Name = "vote";
+  dog.Sound = "big vou vou";
+  IAnimal cat = new Cat();
+  cat.Name = "Rainbow cat";
+  cat.Sound = "big mau mau";
+  IAnimal cow = new Cow();
+  cow.Name = "Holsten";
+
+  Sound sound = new Sound();
+  Console.WriteLine(sound.MakeSound(cow));
+  Console.WriteLine(sound.MakeSound(cat));
+  Console.WriteLine(sound.MakeSound(dog));
+});
+
+
+app.MapGet("/temperature", () =>
+{
+  var temperature = new Temperature(100.4m);
+  Console.WriteLine($" {temperature.Celsius} degree centigrade is equal to : {temperature.GetFahrenheit()} degree farenheight");
+  Console.WriteLine($" {temperature.Celsius} degree farenheight is equal to {temperature.GetCentigrade()} degree centigrade");
+});
+app.MapGet("/exchange", () =>
+{
+  var amount = 200;
+  var exchangeRate = new ExchangeRate("EUR", "DKK");
+  Console.WriteLine(System.Text.Json.JsonSerializer.Serialize(exchangeRate));
+  exchangeRate.Rate = 1.5m;
+  Console.WriteLine($"{amount} {exchangeRate.FromCurrency} is {exchangeRate.Calculate(amount)} {exchangeRate.ToCurrency}");
+});
+app.Run();
+
+//1. cccount
+// Create Account class that can be initialized with the amount value. 
+// Account class contains Withdraw and Deposit methods and Balance (get only) property.
+//  Make sure that you can't withdraw more than you have in the balance.
+public class Account
+{
+  public double Amount { get; set; }
+  public Account(double amount)
+  {
+    Amount = amount;
+  }
+  public double _balance;
+  public double Balance { get => _balance; }
+  public double Deposit(double amountToDeposit)
+  {
+    double result = Amount + amountToDeposit;
```

Comment:
You should also check that amount is not negative

Reply:
i have fixed it thank you